### PR TITLE
Update how Project Filepaths are calculated when not supplied via command line

### PR DESCRIPTION
**Pull request by lumberyard-employee-dm**
Cherry-picking commit https://github.com/o3de/o3de/commit/b9147c60a063ce3c4fda5b6807899b2994605d65 into stabilization branch to allow the change of copying the generated `cmake_dependencies.*.setreg` files to the Cache directory when generating the engine.pak.

Updated how the project runtime filepaths are calculated if project-path parameter is not supplied.
First the "Cache" searching for determining the "project path" has been reverted.
Only scanning upwards for a "project.json" file determines the project root.
Second if a project path can be not be detected, then it defaults to the executable directory(This allow Game Release layout to default the project cache directory to <executable-directory>/Cache).

Also modified the "project cache" detection to search upwards for a "Cache" directory from the executable directory if the `/Amazon/AzCore/Bootstrap/project_cache_path` key isn't set.

### CODE REVIEW

File: Code/Framework/AzGameFramework/AzGameFramework/Application/GameApplication.cpp [hunk 1]
```diff
@@ -82,7 +82,7 @@ namespace AzGameFramework
 
         // Used the lowercase the platform name since the bootstrap.game.<config>.<platform>.setreg is being loaded
```
Comment (AMZN-stankowi) [inline Code/Framework/AzGameFramework/AzGameFramework/Application/GameApplication.cpp:83]: This comment needs to be updated to remove platform, the filename is now just bootstrap.game.<config>.setreg

**Review by lawsonamzn — APPROVED** [review]

File: Code/Framework/AzGameFramework/AzGameFramework/Application/GameApplication.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (lumberyard-employee-dm) [reply Code/Framework/AzGameFramework/AzGameFramework/Application/GameApplication.cpp:83]: Yeah, I'll remove that from the comment.

File: Code/Framework/AzCore/AzCore/Settings/SettingsRegistryMergeUtils.cpp [hunk 2]
```diff
@@ -283,14 +284,18 @@ namespace AZ::SettingsRegistryMergeUtils
...
+        static constexpr AZStd::string_view InternalScanUpProjectRootKey{ "/O3DE/Runtime/Internal/project_root_scan_up_path" };
+        using FixedValueString = SettingsRegistryInterface::FixedValueString;
+        using Type = SettingsRegistryInterface::Type;
+
         AZ::IO::FixedMaxPath projectRoot;
-        const auto projectRootKey = SettingsRegistryInterface::FixedValueString::format("%s/project_path", BootstrapSettingsRootKey);
+        constexpr auto projectRootKey = FixedValueString(BootstrapSettingsRootKey) + "/project_path";
 
-        // Step 1 Run the scan upwards logic once to find the location of the project.json if it exist
+        // Step 1 Run the scan upwards logic once to find the location of the nearset project.json
```
[the remark this answers is not in the document]
Comment (lumberyard-employee-dm) [reply Code/Framework/AzCore/AzCore/Settings/SettingsRegistryMergeUtils.cpp:294]: No, it should be nearest

File: Code/Framework/AzCore/AzCore/Settings/SettingsRegistryMergeUtils.cpp [hunk 3]
```diff
@@ -306,18 +311,147 @@ namespace AZ::SettingsRegistryMergeUtils
...
+            {
+                settingsRegistry.Set(projectCachePathKey, projectCachePath.c_str());
+                // Inject the project root at the end of the command line settings
+                Internal::InjectSettingToCommandLineBack(settingsRegistry, projectCachePathKey, projectCachePath.Native());
+                return projectCachePath;
+            }
+        }
+
+        // Step 3 Append the "Cache" directory to the project-path
+        return projectPath / "Cache";
```
[the remark this answers is not in the document]
Comment (lumberyard-employee-dm) [reply Code/Framework/AzCore/AzCore/Settings/SettingsRegistryMergeUtils.cpp:353]: Can do

**Review by amznestebanpapp — APPROVED** [review]

File: Code/Framework/AzCore/AzCore/Component/ComponentApplication.cpp [hunk 4]
```diff
@@ -216,11 +216,6 @@ namespace AZ
                 m_oldProjectPath = newProjectPath;
 
                 // Merge the project.json file into settings registry under ProjectSettingsRootKey path.
```
Comment (amzn-phist) [inline Code/Framework/AzCore/AzCore/Component/ComponentApplication.cpp:218]: This comment can be removed, the code that does this has been moved elsewhere.

File: Code/Framework/AzCore/AzCore/Settings/SettingsRegistryMergeUtils.cpp [hunk 5]
```diff
@@ -306,18 +313,148 @@ namespace AZ::SettingsRegistryMergeUtils
...
+        constexpr auto projectNameKey = FixedValueString(ProjectSettingsRootKey) + "/project_name";
+
+        // Read the project name from the project.json file if it exists
+        if (AZ::IO::FixedMaxPath projectJsonPath = projectPath / "project.json";
+            AZ::IO::SystemFile::Exists(projectJsonPath.c_str()))
+        {
+            settingsRegistry.MergeSettingsFile(projectJsonPath.Native(),
+                AZ::SettingsRegistryInterface::Format::JsonMergePatch, AZ::SettingsRegistryMergeUtils::ProjectSettingsRootKey);
+        }
+        // Ifa project name isn't set the default will be set to the final path segment of the project path
```
Comment (amzn-phist) [inline Code/Framework/AzCore/AzCore/Settings/SettingsRegistryMergeUtils.cpp:452]: Nit: "If a"

File: Code/Framework/AzCore/AzCore/Settings/SettingsRegistryMergeUtils.h [hunk 6]
```diff
@@ -87,9 +87,9 @@ namespace AZ::SettingsRegistryMergeUtils
     AZ::IO::FixedMaxPath FindEngineRoot(SettingsRegistryInterface& settingsRegistry);
 
     //! The algorithm that is used to find the project root is as follows
-    //! 1. The first time this function is it performs a upward scan for a project.json file from
+    //! 1. The first time this function is it performs an upward scan for a "project.json" file from
```
Comment (amzn-phist) [inline Code/Framework/AzCore/AzCore/Settings/SettingsRegistryMergeUtils.h:90]: Nit: "is" -> "runs"

**Review by amzn-phist — APPROVED** [review]

**Pull request merged**